Type mismatch for Popover

## Describe the bug

When attempting to upgrade from 1.0.0-beta.16 -> 1.0.0 I am getting a Typescript compilation error on `Popover` which I do not understand.

## To Reproduce

Using the following code:

```tsx
      <Overlay
        onHide={() => {
          dispatch(popoverActions.hide(id));
        }}
        placement="bottom"
        rootClose
        show={show}
        target={this.buttonRef.current}
      >
        {({ arrowProps, placement, ref, style }) => {
          return (
            <Popover
              arrowProps={arrowProps}
              className={makeClassName(styles.popover, popoverClassName)}
              id={`${id}-popover`}
              placement={placement}
              ref={ref}
              style={style}
            >
              {content}
            </Popover>
          );
        }}
      </Overlay>
```

generates the following error:

```
src/components/PopoverButton/index.tsx:78:15 - error TS2769: No overload matches this call.
  Overload 1 of 2, '(props: Readonly<ReplaceProps<"div", BsPrefixProps<"div"> & PopoverProps>>): Popover', gave the following error.
    Type 'Record<string, any>' is missing the following properties from type '{ ref: any; style: object; }': ref, style
  Overload 2 of 2, '(props: ReplaceProps<"div", BsPrefixProps<"div"> & PopoverProps>, context?: any): Popover', gave the following error.
    Type 'Record<string, any>' is not assignable to type '{ ref: any; style: object; }'.

78               arrowProps={arrowProps}
                 ~~~~~~~~~~
```

## Additional context

I can see from the Typescript definition files in 1.0.0 that in `Overlay`, `OverlayInjectedProps` includes `arrowProps` with a definition of `Record<string, any>`, and that in `Popover`, `PopoverProps` includes `arrowProps` with a definition of `{ ref: any; style: object }`, which appears to the the mismatch being reported. I don't understand how my code would have any effect on that, i.e., how I can write it to resolve that mismatch, or whether this is in fact a bug in the TS definitions inside this package.

--- a/types/components/Overlay.d.ts
+++ b/types/components/Overlay.d.ts
@@ -22,7 +22,7 @@ export type Placement =
 
 export interface OverlayInjectedProps {
   show: boolean;
-  arrowProps: Record<string, any>;
+  arrowProps: { ref: any; style: object };
   popper: {
     state: any;
     outOfBoundaries: boolean;

